Classical Extras v0.9 (2nd attempt)

MetaTunes opened the pull request:
See readme for full version history

MetaTunes committed:
v0.8

Major update.
Handle multiple recordings and/or multiple parents of a work. Handle
multiple albumartist composers for one track. Option to use "credited
as" name for artists. Option to exclude "solo" from instrument types.
Option to over-ride plugin options with those used when the album was
last saved. Option to keep (and append to) specified existing file tags.
Option to use (and write) SongKong-compatible work tags. Include the
work (and its parents) of which a work is an arrangement. Include
medleys in movement/part description. Allow for multiple parents of
recordings and works (and multiple parents of those). Option as to
whether to include parent works where the relationship attribute is
"part of collection". Plus bug fixes too numerous to mention!

MetaTunes committed:
v0.8+

minor readme updates

MetaTunes committed:
v0.8.1

Bug fixes and minor enhancments

MetaTunes committed:
Minor fix

MetaTunes committed:
v0.8.2

Minor enhancements and bug fixes

MetaTunes committed:
v0.8.4

Major update of UI and functions. See Readme for details.

MetaTunes committed:
Minor changes and readme update

MetaTunes committed:
Readme updates

MetaTunes committed:
Readme update

MetaTunes committed:
Readme updates

MetaTunes committed:
Minor changes

MetaTunes committed:
Readme update

MetaTunes committed:
Update Readme.md

MetaTunes committed:
Minor mods and readme updates

MetaTunes committed:
Readme update

MetaTunes committed:
Readme update

MetaTunes committed:
Readme updates

MetaTunes committed:
Readme update

MetaTunes committed:
Readme updates

MetaTunes committed:
Update UI help

MetaTunes committed:
Version 0.8.5

Bug fixes, mostly

MetaTunes committed:
v0.8.6

More consistent approach to sort tags and hidden variables. Bug fixes.

MetaTunes committed:
v0.8.6

More consistent approach to sort tags and hidden variables. Bug fixes.

MetaTunes committed:
Readme update

MetaTunes committed:
Readme update

MetaTunes committed:
Readme update

MetaTunes committed:
Readme update

MetaTunes committed:
Updated readme and help in UI

MetaTunes committed:
v0.8.7

Various small improvements and bug fixes

MetaTunes committed:
v0.8.8

Fixes for issues #133 and #134

MetaTunes committed:
Readme update

MetaTunes committed:
v0.8.8 for release

Updated v0.8.8 with fixes for #133 & #134

MetaTunes committed:
Readme update

MetaTunes committed:
Minor correction

MetaTunes committed:
Update UI

To reduce size and provide scrollability

MetaTunes committed:
UI update

UI fully scrollable

MetaTunes committed:
v0.8.9

Additional functionality for box sets and other minor
improvements/fixes.

MetaTunes committed:
Fix UI fault

MetaTunes committed:
Updated readme

MetaTunes committed:
Readme update

MetaTunes committed:
v0.9

Additional option to clear previous file tags which are not wanted,
without interfering with cover art. Additional option to replace
instrument names with as-credited names. Also instrument names are now
saved to hidden variables tags (instruments, instruments_credited and
instruments_all) which can be mapped to file tags as required.
Sub-option added to the 'override artist options' option on the
"advanced" tab - to allow tag map details to be included or not in this
over-ride. Minor bug fixes and UI improvements.

MetaTunes committed:
Update to fix Codacy issues

MetaTunes committed:
Fix Codacy issue

MetaTunes committed:
Merge branch '1.0' into 1.0

MetaTunes commented:
Resubmitted PR after resolving conflicts in #139

MetaTunes committed:
Minor change

Remove title case from instruments in genres

MetaTunes commented:
This PR is now ready for review. I don't know why the Travis CI build could not complete - it was OK before I made a very minor change which I don't think could have affected it.

MetaTunes committed:
Remove spurious log statements

MetaTunes commented:
Travis build cancelled again - I've no idea why. Help please!

Sophist-UK commented:
Travis log is empty - it either didn't run at all, or they failed to save the log - so either way most likely to be a problem at Travis' end.

MetaTunes committed:
Re-run autopep8

Very slight tidy-up in the hope that Travis build will proceed

MetaTunes committed:
Revert "Re-run autopep8"

This reverts commit b49f4ce45221a1431523fc7db097d395098e1595.

MetaTunes commented:
A bit of palaver getting through the checks by making 'non-changes' but passed, so I won't attempt any more tweaking!
Just needs review now.

MetaTunes commented:
Not sure if anyone is planning to review this. @zas ?
I appreciate it's quite a big plugin, but it has had a fair bit of testing (see thread at https://community.metabrainz.org/t/classical-extras-plugin/300217 and issues #133, #134, &#135).

Sophist-UK commented on plugins/classical_extras/__init__.py:
```diff
@@ -2,39 +2,42 @@
 PLUGIN_NAME = u'Classical Extras'
 PLUGIN_AUTHOR = u'Mark Evens'
-PLUGIN_DESCRIPTION = u'''This plugin contains 3 classes:
+PLUGIN_DESCRIPTION = u"""This plugin contains 3 classes:
```
The PLUGIN_DESCRIPTION is intended to be used to provide a short user description of what this plugin can do for them rather than an explanation of how it works internally.

Sophist-UK replied:
That is not to say that the current description is not useful to anyone reading the code, though I am finding it very difficult to understand what this code is trying to do given the current level of commenting.

MetaTunes replied:
PLUGIN_DESCRIPTION to be modified in the next commit

Sophist-UK commented on plugins/classical_extras/Readme.md:
```diff
@@ -1,74 +1,147 @@
-1.  The plugin **will not work unless** you have enabled "Use release relationships" and "Use track relationships" in Picard->Options->Metadata.
-2.  It is recommended only to use the plugin on one release at a time, particularly if the "Works and parts" function is being used. The plugin is not designed to do "bulk tagging" of untagged files - use a tool such as SongKong for that and then use the plugin to enhance the results as required. However, once you have tagged files (either in Picard or, say, SongKong) such that they all have Musi ...
-3.  Keep a backup of your picard.ini file (AppData->Roaming->MusicBrainz) in case you erase your settings or Picard crashes and loses them for you.
+1.  The plugin **will not work fully unless** you have enabled "Use release relationships" and "Use track relationships" in Picard->Options->Metadata. However, it may be that the MusicBrainz database has conflicting data between track and release relationships, in which case fix the incorrect data using "Edit relationships" in MusicBrainz.  
```
If this is the case then IMO the plugin should either:

a. Turn these options on, warning the user that it is about to do so;
b. Or give a message saying they need to be turned on before quitting.

MetaTunes replied:
That's a good idea, but there is a slight problem with it, as is implied by the (amended) text: sometimes MB has conflicting metadata between the release and track relationships. Ideally this should be fixed by editing it, but not everyone feels comfortable with that, so I left it for the user to decide which exactly to use. I also deliberately decided not to mess with the 'native' Picard settings. On reflection, it might be best to set the defaults as suggested and to document that (and how to handle exceptions). Will consider for 1.0.

MetaTunes replied:
Latest push sets the options on (no warning, but it is still emphasised in the readme - frankly there's not much point having them off if you want to use the plugin).

Sophist-UK commented on plugins/classical_extras/Readme.md:
```diff
@@ -1,74 +1,147 @@
-2.  It is recommended only to use the plugin on one release at a time, particularly if the "Works and parts" function is being used. The plugin is not designed to do "bulk tagging" of untagged files - use a tool such as SongKong for that and then use the plugin to enhance the results as required. However, once you have tagged files (either in Picard or, say, SongKong) such that they all have Musi ...
-3.  Keep a backup of your picard.ini file (AppData->Roaming->MusicBrainz) in case you erase your settings or Picard crashes and loses them for you.
+1.  The plugin **will not work fully unless** you have enabled "Use release relationships" and "Use track relationships" in Picard->Options->Metadata. However, it may be that the MusicBrainz database has conflicting data between track and release relationships, in which case fix the incorrect data using "Edit relationships" in MusicBrainz.  
+2.  It is recommended only to use the plugin on one or a few release(s) at a time, particularly for initial tagging and if the "Works and parts" function is being used. The plugin is not designed to do "bulk tagging" of untagged files - it may be better to use a tool such as SongKong for that and then use the plugin to enhance the results as required. However, once you have tagged files (either i ...
```
Why can't Picard be used for bulk tagging without this plugin. Why are you recommending SongKong instead?

MetaTunes replied:
Native Picard doesn't tag the work hierarchy, whereas SongKong does. Classical Extras will read the SongKong tags. Another approach would be to get all the MBIDs with native Picard first and then use the plugin to refine files which have the MBID tags. With the latest version this latter approach is now possible without needing two runs (see https://community.metabrainz.org/t/classical-extras-plugin/300217/129) I will test out alternative strategies to see what works best.

Sophist-UK commented on plugins/classical_extras/Readme.md:
```diff
@@ -1,74 +1,147 @@
 
-2. "Name album as 'Composer Last Name(s): Album Name'" will add the composer(s) last name(s) before the album name. MB style is to exclude the composer name unless it is actually part of the album name, but it can be useful to add it for library organisation. The default is checked.
+2. "Work-artist/performer naming options". 
+  This section deals primarily with the application of aliases and credited_as names to replace the MusicBrainz standard names. The first box allows you to choose whether to replace MusicBrainz standard  names by aliases - either for all work-artists/performers or only work-artists (writers, composers, arrangers, lyricists etc.). The second box sets the usage of "as-credited" names: the first par ...
```
This line is over 1000 characters long. MD files will concatenate shorter lines, so this is unnecessary.

Sophist-UK replied:
Any chance that I can have a translation of this paragraph into English so I can understand what it is describing? Or perhaps some examples are needed.

MetaTunes replied:
Unfortunately MusicBrainz is pretty complex - I have attempted to describe how MB works in the box that follows the quoted text (7 different ways an artist can be related to a track!) and even now am not 100% sure it is right. If anyone can point me to a clear documentation of this in MB then I would be most grateful. In addition, the way Picard handles aliases etc. in the context of the "translate" and "standardize" options is inconsistent with the Picard documentation.
Agreed that some examples might help. I'll look into this, but the complexity would require many examples to illustrate the different possibilities.
BTW I've split the para in the next commit.

Sophist-UK commented on plugins/classical_extras/Readme.md:
```diff
@@ -1,74 +1,147 @@
+  There is also option to allow you to replace the track artist by the recording artist (or to merge them). The chosen action will be applied to the 'artist', 'artists', 'artistsort' and 'artists_sort' tags. Note that 'artist' is a single-valued string whereas 'artists' is a list and may be multi-valued. Lists are properly merged, but because the 'artist' string may have different join-phrases et ...
+
+   Note that, if the original track artist is required in tag mapping (i.e. as it was before replacement/merge with recording artist), it is available through the hidden variable _cea_MB_artists.
+
```
We seem to have gone all italic from this point forward. Probably an unclosed MD markup.

MetaTunes replied:
Caused by the underscore. But it displays OK in the browser.

Sophist-UK commented on plugins/classical_extras/Readme.md:
```diff
@@ -1,74 +1,147 @@
+
+  "Modify host tags and include annotations" (Previously called "Include arrangers from all work levels"). This will gather together, for example, any arranger-type information from the recording, work or parent works and place it in the "arranger" tag ('host' tag), with the annotation (see below) in brackets. All arranger types will also be put in a hidden variable, e.g. _cwp_orchestrators. The  ...
+
+   | Artist type | Host tag | Hidden variable |
```
Does this need to go into an MD code block to preserve formatting?

MetaTunes replied:
In the next commit

Sophist-UK commented on plugins/classical_extras/__init__.py:
```diff
@@ -44,24 +47,1444 @@
+    file_options = {}
+    music_file_found = None
+    found_file = False
+    for music_file in album.tagger.files:
```
You do say that this plugin isn't scalable, but this loop is almost certainly part of the reason why, not just the no. of web services calls it makes.

You need to try to process information once as the track is loaded, and not have an n^2 relationship that takes longer and longer to add a file the more files you already have loaded.

MetaTunes replied:
If you can find a solution for this I would be most grateful, but at present I can find no way of directly accessing a filename from within a "track_metadata_processor" - so the only solution was to get the tags of all the files and find a matching track id.
I was hoping to be able to use album.tagger.get_files_from_objects, but that only seems to work at the album level. I guess I may be able to speed it up by just getting the album filenames. 
I agree this could use some improvement. Nevertheless the main time issue is the 1 second constraint on web xml lookups. Even with 2000 tracks, the impact of this loop is small in relation to that inbuilt delay.
BTW, this loop does not increase the number of web service calls as the data is cached - the number of calls is dependent on the total number of works.

Sophist-UK replied:
No - the loop will cause Picard to appear to hang and create a poor user experience.

And I think you may be right about not being able to get to the file from the track metadata - the sequence of events AFAICR is:

* Get or guess the WS Release release MBID from the first file or the cluster.
* Create the album node with this MBID and move the first file to the Unmatched file node.
* Get the WS Release data for the MBID from MB
* Move the rest of the files with that MBID or in the same cluster to the Unmatched files node of the Album
* Process the WS data including calling album and track metadata plugins
* Associate files with tracks (because this uses the metadata to make the track MBID association or guess it.

Unless the WS Get request processes extremely fast, by the time the album / track metadata plugins are run, all the files should be in the Unmatched files node of the Album node and so you should (somehow) be able to access them.

Sophist-UK commented on plugins/classical_extras/Readme.md:
```diff
@@ -282,10 +427,62 @@ Leave the "title" tag unchanged or make it a combination of work and movement.
+To get the whole picture, in XML, for a release, use (for example) https://musicbrainz.org/ws/2/release/f3bb4fdd-5db0-43a8-be73-7a1747f6c2ef?inc=release-groups+media+recordings+artist-credits+artists+aliases+labels+isrcs+collections+artist-rels+release-rels+url-rels+recording-rels+work-rels+recording-level-rels+work-level-rels. This simulates the response given by Picard with the "Use release rel ...
+
+A large variety of releases were used to test this plugin, but there may still be bugs, so further testing is welcome. The following release was particularly complex and useful for testing: https://musicbrainz.org/release/ec519fde-94ee-4812-9717-659d91be11d4. Also this release was a bit tricky - a large box set with some works appearing as originals and in arrangements: https://musicbrainz.org/re ...
+
```
Despite having read this, I am still in the dark about what this plugin really does. I have an idea that it gets more metadata but I have no idea what, or where it is put or how much? Some examples would be very beneficial.

Also I think a warning is needed in the Plugin Description that this plugin is for the dedicated classical music enthusiast and not for the casual or faint-hearted users.

MetaTunes replied:
I have done one example screen showing a result. I agree more could be done. 
> this plugin is for the dedicated classical music enthusiast and not for the casual or faint-hearted users

Classical music enthusiasts are neither casual nor faint-hearted ;)

Sophist-UK commented on plugins/classical_extras/__init__.py:
```diff
@@ -44,24 +47,1444 @@
+    Note regarding aliases and credited-as names:
+    In a MB release, an artist can appear in one of seven contexts. Each of these is accessible in releaseXmlNode
+    and the track and recording contexts are also accessible in trackXmlNode.
+    The seven contexts are:
```
I make this 11 contexts if you count credited-as and alias as separate contexts.

Sophist-UK reviewed (commented):
I have not tried to review the code itself as having read the MD file and the code I have no idea of what it is trying to do or why.

The screen shots in the MD file do give a sense of the breadth of specialised functionality this provides, but it feels like a sink or swim experience if you try to use this.

My overall impression of code quality is, however, that it is good, and given the specialist nature of this plugin, I am less concerned than normal about verifying the functionality.

I am also impressed by the amount of functionality this apparently has - probably more options than Picard native.

My sole concern is that we do not get swamped with support requests by users who have jumped in and find that this is too complicated for them.

I therefore recommend two things:

1. That the MD file is used as the basis for a comprehensive set of Github Wiki pages for this plugin which break it down into smaller more digestible chunks and which provide a more gentle top-down step-by-step explanation of what the plugin is for and what it can do.

2. That this includes many more examples of how this can help users with their classical music.

MetaTunes commented:
Thanks for all the comments, which I will consider more deeply shortly. In the meantime, perhaps some context is achieved by looking at my website: http://music.highmossergate.co.uk/enhanced.php

MetaTunes commented:
Some might say that there is too much documentation. Certainly a lot more than can be said of some other software I won't mention. I think the issue is that the plugin is quite complex and the documentation has evolved as it has grown. BTW I have also added extensive 'tooltips' and 'what's this' in the UI. It really is necessary to use it for a bit in order to understand it (just like Picard). I will look at the comments in a bit more depth when I get a mo.
From a practical point of view, I would like to draw a line at this point (hence the PR), get some more feedback and reflect before making any more changes. After all, I called it v0.9 deliberately. The final version for Picard 1.4 is intended to be v1.0 which will include a bit more functionality but will also address usability issues that arise.

zas commented on plugins/classical_extras/__init__.py:
```diff
@@ -44,24 +47,1444 @@
+    ]
+    tag_options = []
+    for i in range(0, 16):
+        if i < len(default_list):
```
Isn't this equivalent to following code ?

```python
for i in range(0, len(default_list):
```

MetaTunes replied:
No. It does need to run through the whole 16, but there was a theoretical issue regarding undeclared variables, which shouln't occur in practice but which I have nonetheless fixed.

zas commented on plugins/classical_extras/__init__.py:
```diff
@@ -44,24 +47,1444 @@
+        if i < len(default_list):
+            default_source = default_list[i][0]
+            default_tag = default_list[i][1]
+            default_cond = default_list[i][2]
```
```python
default_source, default_tag, default_cond = default_list[i]
```

MetaTunes replied:
Thanks. Still learning Python - far too clever.

zas commented on plugins/classical_extras/__init__.py:
```diff
@@ -44,24 +47,1444 @@
+            default_source = default_list[i][0]
+            default_tag = default_list[i][1]
+            default_cond = default_list[i][2]
+        tag_options.append({'option': 'cea_source_' + unicode(i + 1),
```
You should store the expression `unicode(i +1)` in a variable, and replace all those 6 occurences

zas commented on plugins/classical_extras/__init__.py:
```diff
@@ -44,24 +47,1444 @@
+        names_sort_tag):
+    # Get  artist and apply style
+    for acs in a_list:
+        for ncs in acs:
```
`ncs` and `acs`, hard to know what this means, you should use more explicit variable names

MetaTunes replied:
Agreed. I ran out of ideas. These are only used in a very small context.

zas commented on plugins/classical_extras/__init__.py:
```diff
@@ -44,24 +47,1444 @@
+        if tag:
+            if '~ce_tag_cleared_' + \
+                    tag not in tm or not tm['~ce_tag_cleared_' + tag] == "Y":
+                if tag in tm:
```
It would be better to use a more explicit variable name than `tm`

MetaTunes replied:
I followed the practice of other plugins by using tm as shorthand for track.metadata. I thought it was accepted terminology.

zas commented on plugins/classical_extras/__init__.py:
```diff
@@ -150,54 +1573,137 @@ def get_roman(string):
+    """
+
+    chunks = [a.strip() for a in sortname.split(",")]
+    if len(chunks) == 2:
```
Cache `len(chunks)`

MetaTunes replied:
Will do

Sophist-UK commented:
If you change the PLUGIN_DESCRIPTION to be a more friendly description with warnings to the innocent that would be sufficient IMO for this to be merged so that you can get feedback.

zas reviewed (commented):
This is almost impossible to review:
- code should be splitted in smaller methods (or smaller files)
- comments in the code are lacking, very hard to know what is doing what without digging
- short variable names should only be used for loop indexes
- the patch is huge, the commit history is almost useless because of generic comments

Overall the code is good, and i'm confident it works well enough, but imho the code lacks of clarity, making hard for others to provide improvements and/or fixes.

Also, think about the fact we are moving to python3/qt5 in Picard 2 (with plugins v2), some of the code here will need changes, better to anticipate and make code py2/py3 and qt4/qt5 compatible when it is simple enough. It will ease the transition to plugins v2.

MetaTunes commented:
>If you change the PLUGIN_DESCRIPTION to be a more friendly description

No problems with that - I quite agree!

MetaTunes commented:
I hadn't realised at the time I first did it that the description was for user display rather than developer use.

MetaTunes commented:
For reviewers / approvers: I will add another commit later this evening (which I hope Travis doesn't hiccup over) dealing with documentation issues. The functionality and code will be unchanged.

MetaTunes commented:
>If this is the case then IMO the plugin should either:
>a. Turn these options on, warning the user that it is about to do so;
>b. Or give a message saying they need to be turned on before quitting.

That's a good idea, but there is a slight problem with it, as is implied by the (amended) text: sometimes MB has conflicting metadata between the release and track relationships. Ideally this should be fixed by editing it, but not everyone feels comfortable with that, so I left it for the user to decide which exactly to use. I also deliberately decided not to mess with the 'native' Picard settings. On reflection, it might be best to set the defaults as suggested and to document that (and how to handle exceptions). Will consider for 1.0.

MetaTunes commented:
>Why can't Picard be used for bulk tagging without this plugin. Why are you recommending SongKong instead?

Native Picard doesn't tag the work hierarchy, whereas SongKong does. Classical Extras will read the SongKong tags. Another approach would be to get all the MBIDs with native Picard first and then use the plugin to refine files which have the MBID tags. With the latest version this latter approach is now possible without needing two runs (see https://community.metabrainz.org/t/classical-extras-plugin/300217/129) I will test out alternative strategies to see what works best.

MetaTunes commented:
>This line is over 1000 characters long. MD files will concatenate shorter lines, so this is unnecessary.
>Any chance that I can have a translation of this paragraph into English so I can understand what it is >describing? Or perhaps some examples are needed.

Unfortunately MusicBrainz is pretty complex - I have attempted to describe how MB works in the box that follows the quoted text (7 different ways an artist can be related to a track!) and even now am not 100% sure it is right. If anyone can point me to a clear documentation of this in MB then I would be most grateful. In addition, the way Picard handles aliases etc. in the context of the "translate" and "standardize" options is inconsistent with the Picard documentation.
Agreed that some examples might help. I'll look into this, but the complexity would require many examples to illustrate the different possibilities.

MetaTunes commented:
>We seem to have gone all italic from this point forward. Probably an unclosed MD markup.

Caused by the underscore. But it displays OK in the browser.

Sophist-UK commented:
> Native Picard doesn't tag the work hierarchy, whereas SongKong does. Classical Extras will read the SongKong tags. Another approach would be to get all the MBIDs with native Picard first and then use the plugin to refine files which have the MBID tags. With the latest version this latter approach is now possible without needing two runs (see https://community.metabrainz.org/t/classical-extras-plugin/300217/129) I will test out alternative strategies to see what works best.

No - your plugin should do the SongKong-style tagging - it makes no sense to expect the user to keep switching tagging software.

MetaTunes commented:
Think I may have erased a comment as I moved all my comments into reply boxes (brain fade - a bit late). But I got the point about avoiding multiple tagging software if possible. I need to be comfortable that Picard will handle it now I've got a possible solution - will resolve in 1.0

MetaTunes committed:
Documentation enhancements

MetaTunes committed:
Remove whitespace

MetaTunes commented:
Thanks for the comments @zas 
As per my earlier responses, I agree that some tidying up could be done (and will be done). My plan was to release v0.9, as it works well enough, to get some more usage and feedback. This would be incorporated, along with a lot of code tidying (comments welcome - I'm a fairly newbie Python programmer and learnt on COBOL!) into v1.0 which would then form the basis of moving to Picard 2.0.
Given that context, I hope, with the documentation enhancements suggested by @Sophist-UK , that it is OK to go.

MetaTunes committed:
Non-change

Spurious Codacy error

MetaTunes commented:
No idea why I'm getting the Codacy error
Unused variable '__class__'
3272 class PartLevels:
This happened before then disappeared for no reason. Just making a non-change for now.

Sophist-UK commented:
> No idea why I'm getting the Codacy error
> Unused variable 'class'
> 3272 class PartLevels:
> This happened before then disappeared for no reason. Just making a non-change for now.

If it doesn't disappear, it might be an unmatched curly bracket.

MetaTunes committed:
Revert "Non-change"

This reverts commit e97ea41f4bf8395b08255e4ebdb12033c1da5895.

MetaTunes commented:
>If it doesn't disappear, it might be an unmatched curly bracket.

Surely that would show up as a syntax error? As I said, it has happened before and then disappeared after making a "non-change", so it seems to be a Codacy problem. I'll try something else...

MetaTunes committed:
Modified comments

MetaTunes committed:
Amended comments

MetaTunes commented:
There was one non-matching curly but only in a comment. I fixed it anyway, but the problem persists. Since PartLevels is the main class and it runs fine, the Codacy error is incorrect (or it wouldn't run!). I'll poke it once more - if that fails I will need some (even more) expert help.

MetaTunes committed:
Reversed some odd autopep8 changes

MetaTunes committed:
Address Codacy issues?

MetaTunes committed:
Further codacy fix attempt

MetaTunes commented:
Is this the issue: https://github.com/metabrainz/critiquebrainz/pull/172?
If so, it says it was fixed a few hours ago, but I'm still getting these errors.

MetaTunes committed:
Code tidy

MetaTunes commented:
Can anyone (@zas , @samj1912 ?) help with this?  Codacy is repeatedly failing because it says that the classes are unused variables, which is plainly wrong and seems to be a pylint problem. (Note it also doesn't like the the eval(), but that is not fatal and Picard 1.4 doesn't seem to have ast.)
Thank.

MetaTunes committed:
Code tidy

amending for @zas comments

MetaTunes committed:
Minor changes

Set track-relationships and release-relationships on when loading. + 1
minor fix.

MetaTunes commented:
Still failing Codacy. I'm afraid I have no idea how to resolve this and suspect it is a pylint problem rather than anything wrong with my code. I'd really appreciate help from the experts on this (or at least some comments?). I will have no access to my development machine for the next 2 weeks so can't make any more changes for a while.

MetaTunes committed:
Minor bug fix

MetaTunes commented:
Can a reviewer @zas or @samj1912 help please? I keep getting what appear to be false pylint "unused variable __class__" errors from Codacy. Other than that, this PR should be ready for final review and release. I've fixed most of the previous review issues raised.

mineo commented:
If it's obviously not a bug in your code, just ignore the warnings/errors.

MetaTunes commented:
It is obviously not a bug - if it were a real error, the code would not run at all. I had assumed that the reviewer(s) would not approve it with a Codacy fail, but maybe that assumption is wrong. Whatever, it does need a reviewer to approve it or to tell me why not.

zas commented:
@MetaTunes : ignore the Codacy error. Is the PR ready to merge ?

MetaTunes commented:
@zas : I've just run a test vs a 2,000 track test set and it seems to be error-free. So please merge this and I'll make a start on v1.0. - that is intended to be the final version before migrating to Picard 2.0.

MetaTunes commented:
Have since found a slight bug :(
